### ISSUE TITLE
Proposal: agama.yaml should be packaged seperately from agama-live

### ISSUE THREAD

**Issue by sysrich** [openSUSE/agama#602]
I'm not sure exactly how agama-live is put together, as the documentation seems to reference _service files that are no longer in use

But one thing is clear, there is a single monolithic agama.yaml used by the installer

https://github.com/openSUSE/agama/blob/master/service/etc/agama.yaml

This currently offers "all" the products Agama could install

And then when we want to offer something like an ALP-only flavour, magic is done as part of the kiwi config.sh

```
if [[ "$kiwi_profiles" == "ALP" ]]; then
	filter_config=$(find /usr/lib64/ruby/gems/3.2.0/gems -name filter-config.rb)
	ruby $filter_config /etc/agama.yaml ALP-Bedrock ALP-Micro >/etc/agama.yaml.new
	mv /etc/agama.yaml.new /etc/agama.yaml
fi
```

As fancy as this is, I do not think this will scale, at all

I see a future with multiple different Agama-based installation media all needing different content in their agama.yamls to offer different combinations of products, with different defaults - eg.

all openSUSE Products
all SUSE Products
all ALP Products
all openSUSE immutable desktops (Because the all openSUSE Products installer has too many options)
Some special media for one special Product (that needs more a different set of defaults than the above)
etc etc and so on

doing funky filters and sed's in image builds is not going to be flexible enough for all the different permutations of agama.yaml people are going to want to have for their Products as the adoption grows.

As much as it pains me, I suggest we look to packaging agama.yaml in separate rpms

I propose a naming scheme like "agama-control-$FOO", where $FOO could be the name of a family of products (openSUSE, SUSE, ALP) or the name of a Product, or whatever we need to differentiate it from the other flavours of "agama-control-*"

The packages would include nothing besides the correct agama.yaml for that usecase, and the package would have a "Provides: agama-control" which could then be used as a generic requires on packages like agama-cli so we always make sure that agama has an agama.yaml :)

what do you all think?

**Comment by imobachgs** [comment]
I support this. Some months ago we were discussing about this problem (when we started to build a separate ALP-based ISO), but as we were the only users of Agama, we decided to postpone the implementation. Actually, we even considered having separate files for each product and putting them in `/etc/agama.d`.

Thanks for bringing up this topic again! It looks like we will need to put some effort into this soon.

**Comment by Conan-Kudo** [comment]
Please don't put packaged stuff in `/etc`, though.

**Comment by imobachgs** [comment]
> Please don't put packaged stuff in `/etc`, though.

OK, is `/usr/etc` a better place?

**Comment by Conan-Kudo** [comment]
`/usr/share` please.

**Comment by imobachgs** [comment]
https://trello.com/c/Xcj975dr/

**Comment by jreidinger** [comment]
fixed with #822
